code/modules/robotics/robot/upgrade/mag_boot.dm

```diff
@@ -0,0 +1,24 @@
+/obj/item/roboupgrade/magboot
+    name = "cyborg magnetic traction upgrade"
+    desc = "A set of mag-tractors attached to the underside of a cyborg that emulate magboots."
+    icon_state = "up-mag"
+    drainrate = 10
+    
+/obj/item/roboupgrade/magboot/upgrade_activate(var/mob/living/silicon/robot/user as mob)
+	if (!user)
+		return
+	var/mob/living/silicon/robot/R = user
+	if (!R.part_leg_r && !R.part_leg_l)
```

Comment:
maybe should also check for thruster legs - a little silly to have magboot thrusters